### DISCUSSION TITLE
Parser updates

### DISCUSSION THREAD

**Discussion by luckasRanarison** [nvim-treesitter/nvim-treesitter#5405] [Q&A]
Hi! I'm quite confused with the parser updates, does nvim-treesitter use specific versions from the original parser repository?
For some context, I've encountered some issues with the rust parser, I've checked and nvim-treesitter uses this parser https://github.com/tree-sitter/tree-sitter-rust. 

It wasn't able to parse something like this:
```rust
lazy_static! {
    static ref SOMETHING: Vec<u32> = vec![1, 2, 3];
}
```
TSPlayground Output:
```
macro_invocation [0, 0] - [2, 1]
  macro: identifier [0, 0] - [0, 11]
  token_tree [0, 13] - [2, 1]
    expression_statement [0, 13] - [2, 1]
      block [0, 13] - [2, 1]
        ERROR [1, 4] - [1, 41]
        empty_statement [1, 50] - [1, 51]
    identifier [1, 11] - [1, 14]
    identifier [1, 15] - [1, 24]
    identifier [1, 26] - [1, 29]
    primitive_type [1, 30] - [1, 33]
    identifier [1, 37] - [1, 40]
    token_tree [1, 41] - [1, 50]
      integer_literal [1, 42] - [1, 43]
      integer_literal [1, 45] - [1, 46]
      integer_literal [1, 48] - [1, 49]
```
But I've tried compiling the parser from source and it can correctly parse this expression. I've already updated the parser using `:TSUpdate` and  updated the plugin but it seems like the plugin uses a different version.

**Comment by clason** [comment]
>  I'm quite confused with the parser updates, does nvim-treesitter use specific versions from the original parser repository?

Yes, the ones specified in the `lockfile.json`. The Rust parser is at the latest commit, though, so something else must be going on. `TSUninstall rust` and then install it again.

**Comment by luckasRanarison** [comment]
It's really weird, the version used by nvim-treesitter is 17a6b15562b09db1f27b8f5f26f17edbb2aac097. I've tried compiling and parsing with this version from source using `tree-sitter parse` and it can parse the code above correctly:
```
(source_file [0, 0] - [3, 0]
  (macro_invocation [0, 0] - [2, 1]
    macro: (identifier [0, 0] - [0, 11])
    (token_tree [0, 13] - [2, 1]
      (identifier [1, 11] - [1, 14])
      (identifier [1, 15] - [1, 24])
      (identifier [1, 26] - [1, 29])
      (primitive_type [1, 30] - [1, 33])
      (identifier [1, 37] - [1, 40])
      (token_tree [1, 41] - [1, 50]
        (integer_literal [1, 42] - [1, 43])
        (integer_literal [1, 45] - [1, 46])
        (integer_literal [1, 48] - [1, 49])))))
```
But in neovim, this time with InspectTree but same as TSPlayground:
```
(macro_invocation) ; [1:1 - 3:1]
 macro: (identifier) ; [1:1 - 11]
 (token_tree) ; [1:14 - 3:1]
  (expression_statement) ; [1:14 - 3:1]
   (block) ; [1:14 - 3:1]
    (ERROR) ; [2:5 - 41]
    (empty_statement) ; [2:51 - 51]
  (identifier) ; [2:12 - 14]
  (identifier) ; [2:16 - 24]
  (identifier) ; [2:27 - 29]
  (primitive_type) ; [2:31 - 33]
  (identifier) ; [2:38 - 40]
  (token_tree) ; [2:42 - 50]
   (integer_literal) ; [2:43 - 43]
   (integer_literal) ; [2:46 - 46]
   (integer_literal) ; [2:49 - 49]
```
I've already uninstalled and reinstalled the parser.

**Reply by clason** [reply]
Then it's a Neovim bug.

**Reply by luckasRanarison** [reply]
By the way, jumping to a new line after the "}" in the code above breaks indentation.
